fairscale/nn/pipe/pipe.py

```diff
@@ -371,28 +321,37 @@ def _ensure_copy_streams(self) -> List[List[AbstractStream]]:
 
         return self._copy_streams
 
-    def forward(self, input: TensorOrTensors) -> TensorOrTensors:  # type: ignore
-        """:class:`Pipe` is a fairly transparent module wrapper. It doesn't
+    def forward(self, input) -> RRef:  # type: ignore
+        """
+        Processes a single input mini-batch through the pipe and returns an
+        :class:`~torch.distributed.rpc.RRef` pointing to the output.
+        :class:`Pipe` is a fairly transparent module wrapper. It doesn't
         modify the input and output signature of the underlying module. But
         there's type restriction. Input and output have to be a
-        :class:`~torch.Tensor` or a tuple of tensors. This restriction is
+        :class:`~torch.Tensor` or a sequence of tensors. This restriction is
         applied at partition boundaries too.
 
+        The input tensor is split into multiple micro-batches based on the
+        ``chunks`` parameter used to initialize :class:`Pipe`. The batch size
+        is assumed to be the first dimension of the tensor and if the batch
+        size is less than ``chunks``, the number of micro-batches is equal to
+        the batch size.
+
         Args:
-            input (torch.Tensor or tensors): input mini-batch
+            input (torch.Tensor or sequence of :class:`~torch.Tensor`): input mini-batch
 
         Returns:
-            tensor or tensors: output mini-batch
+            :class:`~torch.distributed.rpc.RRef` to the output of the mini-batch
 
         Raises:
-            TypeError: input is not a tensor or tensors.
+            TypeError: input is not a tensor or sequence of tensors.
 
         """
         microbatch.check(input)
 
         if not self.devices:
             # Empty sequential module is not illegal.
-            return input
+            return RRef(input)
```

Comment:
This would break backward compatibility. Perhaps we could add a constructor parameter to enable the new behavior?